src/explore-education-statistics-frontend/src/modules/download/components/TopicDownloadList.tsx

```diff
@@ -31,22 +32,40 @@ function TopicDownloadList({ topic }: Props) {
             </p>
 
             <ul data-testid={title}>
-              {downloadFiles.map(({ extension, name, path, size }) => (
-                <li key={path}>
-                  <Link
-                    to={`${process.env.CONTENT_API_BASE_URL}/download/${path}`}
-                    data-testid={`download-stats-${path}`}
-                    analytics={{
-                      category: 'Downloads',
-                      action: `Download latest data page ${name} file downloaded`,
-                      label: `File URL: /api/download/${path}`,
-                    }}
-                  >
-                    {name}
-                  </Link>
-                  {` (${extension}, ${size})`}
-                </li>
-              ))}
+              {downloadFiles.map(
+                ({ id: fileId, fileName, extension, name, size }) =>
+                  !fileId && name === 'All files' ? (
```

Comment:
Toggling different elements like this tend to be a bit of a code smell as it doubles the number of changes you have to make (two places instead of one).

Rather than toggling, we could clean this up to only use a single element. Something like this:

```tsx
const isAllFile = fileId && name === 'All files';

const url = `${process.env.CONTENT_API_BASE_URL}/releases/${latestReleaseId}/files/${
  isAllFile ? 'all' : fileId
}`

return (
  <Link
    to={url}
    analytics={{
      ...
      action: `Download latest data page ${isAllFiles ? 'all files' : 'file'} downloaded`
    }}
    ...
  />
);
````

Reply:
Thanks that's a good idea, I've done that now.